packages/contracts/utils/storage.ts

```diff
@@ -0,0 +1,26 @@
+import {ethers} from 'hardhat';
+import {defaultAbiCoder} from 'ethers/lib/utils';
+
+import {IMPLEMENTATION_SLOT} from '../test/test-utils/uups-upgradeable';
+
+export async function readImplementationValuesFromSlot(
+  contractAddresses: string[]
+): Promise<string[]> {
+  const implementationValues: string[] = await Promise.all(
+    contractAddresses.map(async contractAddress => {
+      return readImplementationValueFromSlot(contractAddress);
+    })
+  );
+
+  return implementationValues;
```

Comment:
Very minor, but we don't really need any async features code here. Plain promises are simpler and lighter

```suggestion
export function readImplementationValuesFromSlot(
  contractAddresses: string[]
): Promise<string[]> {
  return Promise.all(
    contractAddresses.map(contractAddress =>
      readImplementationValueFromSlot(contractAddress)
    )
  );
```